src/hooks.spec.tsx

```diff
@@ -447,44 +451,33 @@ describe('hooks', () => {
           <MyFeatureComponent />
         </OptimizelyProvider>
       );
-      await optimizelyMock.onReady();
       await waitFor(() => expect(screen.getByTestId('result')).toHaveTextContent('false|{}|true|false'));
     });
 
     it('should respect the timeout option passed', async () => {
+      mockDelay = 100;
       isFeatureEnabledMock.mockReturnValue(false);
       featureVariables = {};
       readySuccess = false;
 
       render(
         <OptimizelyProvider optimizely={optimizelyMock}>
-          <MyFeatureComponent options={{ timeout: mockDelay }} />
+          <MyFeatureComponent options={{ timeout: mockDelay - 10 }} />
         </OptimizelyProvider>
       );
 
-      await waitFor(() => expect(screen.getByTestId('result')).toHaveTextContent('false|{}|false|false')); // initial render
-
-      await optimizelyMock.onReady();
-      await waitFor(() => expect(screen.getByTestId('result')).toHaveTextContent('false|{}|false|true')); // when didTimeout
+      // Initial render
+      expect(screen.getByTestId('result')).toHaveTextContent('false|{}|false|false');
 
-      // Simulate datafile fetch completing after timeout has already passed
-      // isFeatureEnabled now returns true, getFeatureVariables returns variable values
+      readySuccess = true;
       isFeatureEnabledMock.mockReturnValue(true);
       featureVariables = mockFeatureVariables;
-      // Wait for completion of dataReadyPromise
-      await optimizelyMock.onReady().then((res) => res.dataReadyPromise);
 
-      // Simulate datafile fetch completing after timeout has already passed
-      // Activate now returns a variation
-      activateMock.mockReturnValue('12345');
-      // Wait for completion of dataReadyPromise
-      await optimizelyMock.onReady().then((res) => res.dataReadyPromise);
-      await waitFor(() => expect(screen.getByTestId('result')).toHaveTextContent('true|{"foo":"bar"}|true|true')); // when clientReady
+      // When timeout is reached, but dataReadyPromise is resolved later with the feature enabled
+      await waitFor(() => expect(screen.getByTestId('result')).toHaveTextContent('true|{"foo":"bar"}|true|true'));
     });
 
     it('should gracefully handle the client promise rejecting after timeout', async () => {
-      jest.useFakeTimers();
```

Comment:
Cool. Tests still pass. Are Jest's fake timers 👎 nowadays?

Reply:
No. They are same as before. Its just these tests do not require this functionality anymore. I do not know why its here for in the first place. It might have been useful in the past, but for current suite its not required anymore.